habitat_baselines/rl/ppo/ppo_trainer.py

```diff
@@ -520,27 +867,33 @@ def _eval_checkpoint(
         batch = apply_obs_transforms_batch(batch, self.obs_transforms)
 
         current_episode_reward = torch.zeros(
-            self.envs.num_envs, 1, device=self.device
+            self.envs.num_envs, 1, device="cpu"
```

Comment:
why do we want to allocate rewards, masks and etc on cpu?

Reply:
The little update operations that we do with them are much faster on the CPU than GPU because the GPU is total overkill for adding a handful over numbers together.

Reply:
Got it. Was it measured or based on experience?

Reply:
Measured.  It made more of a difference with the work with Brennan (this was actually a non-trivial bottleneck there), here it helped just a little bit (couple FPS for pointnav) but hey, why not!